File: score-http/score-http/src/main/java/org/oagi/score/gateway/http/api/tag_management/data/Tag.java
package org.oagi.score.gateway.http.api.tag_management.data;

import lombok.Data;

import java.math.BigInteger;
import java.time.LocalDateTime;

@Data
public class Tag {

    private BigInteger tagId;

    private String name;

    private String description;

    private String textColor;

    private String backgroundColor;

    private BigInteger createdBy;

    private BigInteger lastUpdatedBy;

    private LocalDateTime creationTimestamp;

    private LocalDateTime lastUpdateTimestamp;

}

File: connect-center-api/app/repositories/models/tag.py
"""Repository row model for tag resources."""


from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class TagRow(BaseModel):
    """Represent TagRow."""
    tag_id: int
    name: str
    description: str | None = None
    color: str | None = None
    text_color: str | None = None
    created_by: int
    last_updated_by: int
    creation_timestamp: datetime
    last_update_timestamp: datetime

    model_config = ConfigDict(frozen=True, from_attributes=True)
